src/main/resources/static/vulnerableApp.js

```diff
@@ -160,14 +170,29 @@ function handleElementAutoSelection(vulnerableAppEndPointData, id = 0) {
   }
 
   if (id === 0) {
-    detailTitle.innerHTML = vulnerableAppEndPointData[id]["Description"];
-  } else {
+// Assume detailTitle is a reference to a DOM element
+var detailTitle = document.getElementById('detailTitle');
+
+// Fetch the user-controlled data from the endpoint
+var description = vulnerableAppEndPointData[id]["Description"];
+
+// Sanitize the user-controlled data to remove any potentially malicious scripts
+// You can use a library like DOMPurify to sanitize the content
+var sanitizedDescription = DOMPurify.sanitize(description);
+
+// Use textContent or innerText instead of innerHTML to prevent execution of any scripts
+detailTitle.textContent = sanitizedDescription;
     innerMaster.innerHTML = "";
   }
 
   vulnerabilitySelected = vulnerableAppEndPointData[id]["Name"];
-  detailTitle.innerHTML = vulnerableAppEndPointData[id]["Description"];
```

Comment:
CWE-79: Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')
------
User controlled data in methods like `innerHTML`, `outerHTML` or `document.write` is an anti-pattern that can lead to XSS vulnerabilities